core/types/src/error.rs

```diff
@@ -34,7 +35,7 @@ pub enum FfiError {
 ///  9. `0x8000-0x8fff`: Attestation key errors.
 /// 10. `0xf000-0xffff`: Internal (to SGX) errors.
 #[non_exhaustive]
-#[derive(Copy, Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+#[derive(Copy, Clone, Debug, displaydoc::Display, Eq, Hash, PartialEq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
Should we be using `#[prefix_enum_doc_attributes]` on these types?

Reply:
I don't think in this case, unless we wanted to rework the initial doc comment
> combines the doc comment message on your enum itself with the messages for each variant

with this test:
 ```rust
     #[test]
    fn display_enum() {
        assert_eq!(Error::Unexpected.to_string(), "taco")
    }
```
the result is
```
thread 'error::test::display_enum' panicked at 'assertion failed: `(left == right)`
  left: `"A enumeration of SGX errors.\n\nThose listed here are the ones which are identified in the `sgx_status_t`\nenum, in order of the actual value. Note that values are grouped\n(numerically) into the following general sections:\n\n1. `0x0000-0x0fff`: Generic errors.\n2. `0x1000-0x1fff`: Fatal runtime errors.\n3. `0x2000-0x2fff`: Enclave creation errors.\n4. `0x3000-0x3fff`: Local attestation/report verification errors.\n5. `0x4000-0x4fff`: Errors when communicating with the Architectural\nEnclave Service Manager (AESM).\n6. `0x5000-0x5fff`: Errors internal to AESM.\n7. `0x6000-0x6fff`: Errors with the encrypted enclave loader.\n8. `0x7000-0x7fff`: Errors with the \"SGX Encrypted FS\" utility.\n9. `0x8000-0x8fff`: Attestation key errors.\n10. `0xf000-0xffff`: Internal (to SGX) errors.: An unexpected error."`,
 right: `"taco"`', core/types/src/error.rs:382:9
 ```